test/integration/e2e_test.go

```diff
@@ -0,0 +1,117 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/linode/linodego"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/oauth2"
+)
+
+const (
+	packerTemplate = "template/test_image_template.json"
+)
+
+func TestBuildPackerImage(t *testing.T) {
+	linodeToken := os.Getenv("LINODE_TOKEN")
+
+	if linodeToken == "" {
+		t.Fatal("Linode token is not set. Please set LINODE_TOKEN as environment variable.")
+	}
+
+	// Run the Packer build command from terminal
+	cmd := exec.Command("packer", "build", packerTemplate)
+	output, err := cmd.CombinedOutput()
+	// Check if the Packer build was successful
+	if err != nil {
+		t.Fatalf("Error building Packer image: %v\nOutput:\n%s", err, output)
+	}
+
+	// Assert the output contains expected strings
+	expectedSubstring := "Builds finished. The artifacts of successful builds"
+	assert.True(t, strings.Contains(string(output), expectedSubstring), "Expected successful build output to contain: %s", expectedSubstring)
+
+	// Assert other fields
+	err = assertLinodeImage("test-packer-image-", t)
+
+	if err != nil {
+		t.Fatalf("Error asserting Linode builder image: %v", err)
+	}
+
+	defer func() {
+		if err := teardown(); err != nil {
+			fmt.Printf("Error during deleting image after test execution: %v\n", err)
+		}
+	}()
```

Comment:
Can you place the `defer` block above to ensure it always be executed? (Otherwise, if the test error out earlier, then defer teardown won't be executed before return)